packages/@aws-cdk/aws-eks/lib/managed-nodegroup.ts

```diff
@@ -0,0 +1,278 @@
+import { InstanceType, ISecurityGroup, SubnetSelection } from '@aws-cdk/aws-ec2';
+import { IRole, ManagedPolicy, Role, ServicePrincipal } from '@aws-cdk/aws-iam';
+import { CfnOutput, Construct, IResource, Resource } from '@aws-cdk/core';
+import { Cluster } from './cluster';
+import { CfnNodegroup } from './eks.generated';
+
+/**
+ * NodeGroup interface
+ */
+export interface INodegroup extends IResource {
+  /**
+   * Name of the nodegroup
+   * @attribute
+   */
+  readonly nodegroupName: string
+}
+
+/**
+ * The AMI type for your node group. GPU instance types should use the `AL2_x86_64_GPU` AMI type, which uses the Amazon EKS-optimized
+ * Linux AMI with GPU support. Non-GPU instances should use the `AL2_x86_64` AMI type, which uses the Amazon EKS-optimized Linux AMI.
+ */
+export enum NodegroupAmiType {
+  /**
+   * Amazon EKS-optimized Linux AMI
+   */
+  AL2_X86_64 = 'AL2_x86_64',
+  /**
+   *  Amazon EKS-optimized Linux AMI with GPU support
+   */
+  AL2_X86_64_GPU = 'AL2_x86_64_GPU',
+}
+
+/**
+ * The remote access (SSH) configuration to use with your node group.
+ *
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-eks-nodegroup-remoteaccess.html
+ */
+export interface NodegroupRemoteAccess {
+  /**
+   * The Amazon EC2 SSH key that provides access for SSH communication with the worker nodes in the managed node group.
+   */
+  readonly ec2SshKey: string;
+  /**
+   * The security groups that are allowed SSH access (port 22) to the worker nodes. If you specify an Amazon EC2 SSH key but do not
+   * specify a source security group when you create a managed node group, then port 22 on the worker nodes is opened to the internet (0.0.0.0/0).
+   *
+   * @default - port 22 on the worker nodes is opened to the internet (0.0.0.0/0)
+   */
+  readonly sourceSecurityGroups?: ISecurityGroup[];
+}
+
+/**
+ * The Nodegroup Options for addNodeGroup() method
+ */
+export interface NodegroupOptions {
+  /**
+   * Name of the Nodegroup
+   *
+   * @default - resource ID
+   */
+  readonly nodegroupName?: string;
+  /**
+   * The subnets to use for the Auto Scaling group that is created for your node group. These subnets must have the tag key
+   * `kubernetes.io/cluster/CLUSTER_NAME` with a value of `shared`, where `CLUSTER_NAME` is replaced with the name of your cluster.
```

Comment:
I am assuming this is automatically happening when the VPC is defined in the same CDK app. Correct? Worth mentioning.